src/main/webapp/app/exam/participate/exercises/exam-exercise-update-highlighter/exam-exercise-update-highlighter.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { ExamExerciseUpdateService } from 'app/exam/manage/exam-exercise-update.service';
+import { Exercise } from 'app/entities/exercise.model';
+import { DiffMatchPatch } from 'diff-match-patch-typescript';
+
+@Component({
+    selector: 'jhi-exam-exercise-update-highlighter',
+    templateUrl: './exam-exercise-update-highlighter.component.html',
+    styleUrls: ['./exam-exercise-update-highlighter.component.scss'],
+})
+export class ExamExerciseUpdateHighlighterComponent implements OnInit {
+    subscriptionToLiveExamExerciseUpdates: Subscription;
+    previousProblemStatementUpdate: string;
+    updatedProblemStatementWithHighlightedDifferences: string;
+    updatedProblemStatement: string;
+    showHighlightedDifferences = true;
+
+    @Input() exercise: Exercise;
+
+    @Output() problemStatementUpdateEvent: EventEmitter<string> = new EventEmitter<string>();
+
+    constructor(private examExerciseUpdateService: ExamExerciseUpdateService) {}
+
+    ngOnInit(): void {
+        this.subscriptionToLiveExamExerciseUpdates = this.examExerciseUpdateService.currentExerciseIdAndProblemStatement.subscribe((update) => {
+            this.updateExerciseProblemStatementById(update.exerciseId, update.problemStatement);
+        });
+    }
+
+    toggleHighlightedProblemStatement(): void {
+        if (this.showHighlightedDifferences) {
+            this.exercise.problemStatement = this.updatedProblemStatement;
+        } else {
+            this.exercise.problemStatement = this.updatedProblemStatementWithHighlightedDifferences;
+        }
+        this.showHighlightedDifferences = !this.showHighlightedDifferences;
+        this.problemStatementUpdateEvent.emit(this.exercise.problemStatement);
+    }
+
+    updateExerciseProblemStatementById(exerciseId: number, updatedProblemStatement: string) {
+        if (updatedProblemStatement != undefined && exerciseId === this.exercise.id) {
+            this.updatedProblemStatement = updatedProblemStatement;
+            this.exercise.problemStatement = this.highlightProblemStatementDifferences();
+        }
+        this.problemStatementUpdateEvent.emit(this.exercise.problemStatement);
+    }
+
+    highlightProblemStatementDifferences() {
```

Comment:
I usually like to add JS style comments for the function definitions on the client side as well. I assumed that was also a standard. It would be nice to add them here. Maybe also in the functions above. If you think it makes sense?

Reply:
Good idea. Is done 👍